lib/methods/veres-one/veres-one.js

```diff
@@ -0,0 +1,375 @@
+    // }
+    const {didDocument} = options;
+
+    if(options.accelerator) {
+      // send operation to an accelerator for proof
+      operation = await this.attachAcceleratorProof({operation, ...options});
+    } else {
+      // attach an equihash proof
+      // _log(options, 'Generating Equihash proof of work... (60-120 seconds)');
+      operation = await this.attachEquihashProof({operation});
+    }
+
+    // get private key
+    const invokeKeyId = didDocument.doc.invokeCapability[0].publicKey[0].id;
+    const creator = invokeKeyId;
+    const invokeKey = didDocument.keys[invokeKeyId];
+    if(!invokeKey.secretKey) {
+      throw new Error('Invocation key required to perform a send');
+    }
+
+    const secretKey = await invokeKey.export();
+
+    // attach capability invocation proof
+    // _log(options, 'Attaching LD-OCAP invocation proof...');
+    const capabilityAction = operation.type.startsWith('Create')
+      ? 'RegisterDid'
+      : 'UpdateDidDocument';
+
+    operation = await this.attachInvocationProof({
+      operation,
+      capability: didDocument.id,
+      capabilityAction,
+      creator,
+      privateKeyPem: secretKey.secretKeyPem,
+      privateKeyBase58: secretKey.secretKeyBase58,
+    });
+
+    // send DID Document to a Veres One ledger node
+    // if (operationType === 'create') {
+    //   _log(options, 'Registering DID on Veres One...');
+    // } else {
+    //   _log(options, 'Updating DID Document on Veres One...');
+    // }
+    const response = await this.client
+      .send({operation, authKey: invokeKey, ...options});
+
+    if(response.status !== 204) {
+      // TODO: Throw error here
+      // _error(options, 'Failed to register DID Document.');
+      // _error(options, 'Status Code: ' + response.status);
+      // _error(options, 'Response Body: ' +
+      //   JSON.stringify(await response.json(), null, 2));
+    }
+    // if(operationType === 'create') {
+    //   _log(options, 'DID registration send to ledger.');
+    // } else {
+    //   _log(options, 'DID Document update sent to the Veres One ledger.');
+    // }
+    // _log(options, 'Please wait ~15-30 seconds for ledger consensus.');
+    // _log(options, 'You may use the `info` command to monitor the ' +
+    //   'registration of your DID.');
+
+    if(options.notes) {
+      // save ledger if requested
+      this.meta.saveNotes(didDocument, options);
+    }
+    return response;
+  }
+
+  /**
+   * Sends a ledger operation to an accelerator.
+   * Required when registering a DID Document (and not using an Equihash proof).
+   *
+   * @param options {object}
+   *
+   * @returns {Promise<object>} Resolves with a wrapped Web Ledger operation,
+   *   with additional proof from accelerator.
+   */
+  async attachAcceleratorProof(options) {
+    if(!options.authDoc) {
+      throw new Error('Authorization DID Doc required');
+    }
+    const authKey = options.authDoc.doc.authentication[0];
```

Comment:
has the options stuff been validated to the point you can apply this level of specificity without risking an uncaught error?

Reply:
Thanks, should be addressed now.